Add 'add_version_condition' arg

ikonst opened the pull request:
Add a flag for controlling whether `Model.save`, `Model.update` and `Model.delete` add a condition that the persisted version is the same as the in-memory one, defaulting to `True` (the "safe" behavior)

Fixes #1168.

garrettheel reviewed (approved):
Looks like a reasonable way to allow for granular overrides. Should we just add this to `save` for consistency as well? Given that there may sometimes be a use-case to totally disregard the version check there too

ikonst commented on pynamodb/attributes.py:
```diff
@@ -772,7 +772,36 @@ def deserialize(self, value):
 class VersionAttribute(NumberAttribute):
     """
-    A version attribute
+    A number attribute that denotes a version / revision counter of the model.
```
TODO simply say it's an optimistic locking mechanism and move the prose to a separate page

ikonst commented:
@garrettheel What you reviewed before was totally broken -- when `False` it didn't add condition **nor** incremented version 😅 

This time
- fixed the thing
- updated tests
- removed the prose from VersionAttribute, instead linking to an "optimistic locking" page that we already had, and adding a section to it about skipping the condition
